app/gui/src/presenter/graph/state.rs

```diff
@@ -454,14 +454,19 @@ impl<'a> ControllerChange<'a> {
         let new_displayed_expr = node_view::Expression {
             pattern:             node.info.pattern().map(|t| t.repr()),
             code:                node.info.expression().repr().into(),
-            whole_expression_id: node.info.expression().id,
+            whole_expression_id: Some(node.info.id()),
             input_span_tree:     trees.inputs,
             output_span_tree:    trees.outputs.unwrap_or_else(default),
         };
         let mut nodes = self.nodes.borrow_mut();
         let displayed = nodes.get_mut_or_create(ast_id);
 
-        if displayed.expression != new_displayed_expr {
+        let displayed_updated = displayed.expression != new_displayed_expr;
+        let context_switch_updated = displayed.context_switch != node.info.ast_info.context_switch;
+        let skip_updated = displayed.is_skipped != node.info.macros_info().skip;
+        let freeze_updated = displayed.is_frozen != node.info.macros_info().freeze;
+
+        if displayed_updated || context_switch_updated || skip_updated || freeze_updated {
```

Comment:
@farmaazon I noticed that the button's highlight is not updated when I reopen the project. I spent some time debugging, and we had a bug here for over six months. Since I changed this code, we compare only displayed code and completely ignore SKIP and FREEZE macros. It might be the reason for issues reported about macros a long time ago before we disabled them.

Now it is fixed, and the state of all buttons is updated correctly. Well, SKIP and FREEZE are broken in the parser, so it does not matter for them.